app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class Api::V1::BaseController < ActionController::API
+  include Pundit
+  include CustomErrors
+
+  rescue_from ActionController::ParameterMissing do
+    api_error(status: 400, errors: "invalid parameters")
+  end
+
+  rescue_from Pundit::NotAuthorizedError do
+    unauthorized!
+  end
+
+  rescue_from ActiveRecord::RecordNotFound do
+    api_error(status: 404, errors: "resource not found!")
+  end
+
+  rescue_from ActiveRecord::RecordInvalid do
+    api_error(status: 422, errors: "resource invalid!")
+  end
+
+  rescue_from UnauthenticatedError do
+    unauthenticated!
+  end
+
+  def authenticate_user
+    header = request.headers["Authorization"]
+    header = header.split(" ").last if header
+    begin
+      @decoded = JsonWebToken.decode(header)
+      @current_user = User.find(@decoded[:user][:user_id])
+    rescue JWT::DecodeError => e
+      api_error(status: 401, errors: e.message)
+    end
+  end
+
+  def authenticate_user!
+    authenticate_user or raise UnauthenticatedError
+  end
+
+  def unauthenticated!
+    api_error(status: 401, errors: "not authenticated")
+  end
+
+  def unauthorized!
+    api_error(status: 403, errors: "not authorized")
+  end
+
+  def not_found!
+    api_error(status: 404, errors: "resource not found")
+  end
+
+  def invalid_resource!(errors = [])
+    api_error(status: 422, errors: errors)
+  end
+
+  def paginate(resource)
+    default_per_page = Rails.application.secrets.default_per_page || 5
```

Comment:
default_per_page is not a secret

Reply:
Should i remove it or we can add this as a secret..

Reply:
Make it a constant